core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1519,7 +1519,12 @@ class SparkContext(config: SparkConf) extends Logging {
   def addFile(path: String, recursive: Boolean): Unit = {
     val uri = new Path(path).toUri
     val schemeCorrectedPath = uri.getScheme match {
-      case null | "local" => new File(path).getCanonicalFile.toURI.toString
+      case null => new File(path).getCanonicalFile.toURI.toString
+      case "local" =>
+        logWarning("We do not support add a local file here because file with local scheme is " +
+          "already existed on every node, there is no need to call addFile to add it again. " +
+          "(See more discussion about this in SPARK-24195.)")
```

Comment:
Can we please rephrase to "File with 'local' scheme is not supported to add to file server, since it is already available on every node."?

Reply:
Got it, rephrase done.